class-unlist-posts.php

```diff
@@ -128,7 +128,9 @@ function post_navigation_clause( $where ) {
 
 			// bail if none of the posts are hidden or we are on admin page or singular page.
 			if ( ( is_admin() && ! wp_doing_ajax() ) || in_array( get_the_ID(), $hidden_posts, true ) || empty( $hidden_posts ) ) {
-				return $where;
+				if ( ( is_admin() && ! wp_doing_ajax() ) || empty( $hidden_posts ) ) { // condition run when user already on unlisted post.
```

Comment:
Instead of double if conditions, Can we simply update the first if condition?

Reply:
Yes, we can. Updated

Reply:
This change is not working as expected - can you debug this?
I have also merged unit tests in this branch which is failing as this fix is not working.